src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -107,17 +106,24 @@ public void GenSspiClientContext(SspiClientContextStatus sspiClientContextStatus
             ContextFlagsPal contextFlags = sspiClientContextStatus.ContextFlags;
             SafeFreeCredentials credentialsHandle = sspiClientContextStatus.CredentialsHandle;
 
+            string securityPackage = NegotiationInfoClass.Negotiate;
+
+            if (securityContext == null)
+            {
+                credentialsHandle = NegotiateStreamPal.AcquireDefaultCredential(securityPackage, false);
+            }
+
             SecurityBuffer[] inSecurityBufferArray = null;
-            if (securityContext == null) //first iteration
```

Comment:
When `receivedBuff == null`, we don't want to create `SecurityBuffer` with `receivedBuff`. In that case, we just need empty `SecurityBuffer` array. When I was looking at code for how to Integrated Authentication works in `System.Net.Security` I found there is no guaranty that `receivedBuff` is not `null` when there is existing `credentialsHandle` obtained by previous iteration. So this fix is more appropriate to be safer.